### PR TITLE
WSTEAM1-1312 Run an Optimizely AA Test

### PR DESCRIPTION
**Pull request by emilysaffron**
Resolves JIRA [WSTEAM1-1312](https://jira.dev.bbc.co.uk/browse/WSTEAM1-1312)

Overall changes
======

Sets up a basic AA test for the srollablePromo component on an ArticlePage. This test has been set up in optimizely to only run on the swahili service. Both variations will return the same component.

Code changes
======

- Wraps the article page in the optimizely provider
- Applys the `useOptimizelyVariation` hook to the scrollable promo

Testing
======
1. _List the steps used to test this PR._

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

### CODE REVIEW

File: src/app/pages/ArticlePage/index.tsx
Comment (emilysaffron) [inline src/app/pages/ArticlePage/index.tsx:1]: just seen that this is done in https://github.com/bbc/simorgh/pull/11997 !  will leave it for now and resolve merge conflicts once the other PR is merged

File: src/app/pages/ArticlePage/ArticlePage.tsx
Comment (karinathomasbbc) [inline src/app/pages/ArticlePage/ArticlePage.tsx:182]: I don't know a huge amount about optimizely, but I assume that if we had an AB test, then we would do something different for 1 variant. What if we did nothing here (i.e. revert the changes to this line), would it still work? I suppose we'll never really do AA tests - this is just to prove that we get the expected results in Optimizely, right?

**Review by karinathomasbbc — APPROVED** [review]

File: src/app/pages/ArticlePage/ArticlePage.tsx
Comment (emilysaffron) [reply src/app/pages/ArticlePage/ArticlePage.tsx:182]: Yeah this is just to prove that we get metrics through for both variations we've set up. I also dont know much about optimizely lol but assume yea if this was an AB test we'd render a different component depending on the variant. Good Q about doing nothing there, i think yes it would still work as the variant is calculated by optimizely based off of a userId set with a cookie (will defer to @HarveyPeachey to fact check me there though), but good practice to leave it there for clarity and testing's sake

**Review by amoore108 — APPROVED** [review]
Looks good 👍 Presume it hinges on https://github.com/bbc/simorgh/pull/11997 being merged first?

File: src/app/pages/ArticlePage/ArticlePage.tsx
Comment (karinathomasbbc) [reply src/app/pages/ArticlePage/ArticlePage.tsx:182]: Could/should we do something like wrap the A test variant in a div with the variation ID, or is that overkill?

e.g. 

```suggestion
    links: (props: ComponentToRenderProps) =>
      scrollablePromoVariation === 'variation_1_aa' ? (
        <div data-testid={scrollablePromoVariation}>
           <ScrollablePromo {...props} />
        </div>
      ) : (
        <ScrollablePromo {...props} />
      ),
```

Comment (emilysaffron) [reply src/app/pages/ArticlePage/ArticlePage.tsx:182]: would this be so we can see it coming through in dev tools when we visit it ? i think that could be overkill,  we'll be tracking what variants come through to users through optimizely, so dont think we need to be able to see the variant id ourselves - if it's not working we'll be able to tell bc no metrics will come through

**Comment by emilysaffron** [comment]
> Looks good 👍 Presume it hinges on #11997 being merged first?

Yep will hold off on merging until that goes in 😊

File: src/app/pages/ArticlePage/ArticlePage.tsx
Comment (karinathomasbbc) [reply src/app/pages/ArticlePage/ArticlePage.tsx:182]: Yeah thought we could add a unit test for it, but perhaps it is overkill. I've approved anyway, so feel free to merge when it's ready to go!

Comment (HarveyPeachey) [reply src/app/pages/ArticlePage/ArticlePage.tsx:182]: Yeah AA tests are only for testing this, as Emily said, if it was an a proper experiment we would see something else render based on variation. And yeah this is just to prove everything is working

File: src/app/pages/ArticlePage/ArticlePage.tsx
Comment (HarveyPeachey) [inline src/app/pages/ArticlePage/ArticlePage.tsx:142]: One thing to check is the problem we had a while ago https://github.com/bbc/simorgh/pull/9891 that @amoore108 fixed, if we can check http://localhost:7080/mundo/23263889 to make sure it doesn't break. If it does I think this can be fixed by using this within the `ScrollablePromo` component, as useOptimizelyVariation hook uses state, it will re-render the whole `ArticlePage` component, but if the hook is within `ScrollablePromo` I think it will only re-render that section of the DOM.

I think in the future we need to think about making a component maybe to hold this hook so state is encapsulated away from the page component

Comment (emilysaffron) [reply src/app/pages/ArticlePage/ArticlePage.tsx:142]: good shout, just checked it and all the includes and everything seem to be rendering correctly 👍

Comment (amoore108) [reply src/app/pages/ArticlePage/ArticlePage.tsx:142]: Yea good shout @HarveyPeachey , had forgotten about that previous behaviour

**Review by HarveyPeachey — COMMENTED** [review]
I think we need to add in all the metric mechanisms, as I think they were [removed](https://github.com/bbc/simorgh/pull/11997/commits/9830bdbdd3025359de4fb342e2352e2ccda3df11#diff-7b748265dea863c1d16b9d5339803ce5a801ecbb8584ae09860827e0de3a6848) in #11997, also need to setup the click and view tracking for the component with optimizely.

Happy to pair on it to get it working

**Comment by emilysaffron** [comment]
> I think we need to add in all the metric mechanisms, as I think they were [removed](https://github.com/bbc/simorgh/pull/11997/commits/9830bdbdd3025359de4fb342e2352e2ccda3df11#diff-7b748265dea863c1d16b9d5339803ce5a801ecbb8584ae09860827e0de3a6848) in #11997, also need to setup the click and view tracking for the component with optimizely.
> 
> Happy to pair on it to get it working

by metric mechanisms do you mean like the scroll depth and article complete?

**Comment by HarveyPeachey** [comment]
> > I think we need to add in all the metric mechanisms, as I think they were [removed](https://github.com/bbc/simorgh/pull/11997/commits/9830bdbdd3025359de4fb342e2352e2ccda3df11#diff-7b748265dea863c1d16b9d5339803ce5a801ecbb8584ae09860827e0de3a6848) in #11997, also need to setup the click and view tracking for the component with optimizely.
> > Happy to pair on it to get it working
> 
> by metric mechanisms do you mean like the scroll depth and article complete?

Yeah and Page view tracking, also component [click](https://github.com/bbc/simorgh/tree/latest/src/app/hooks/useClickTrackerHandler#:~:text=A%20click%20event%20is%20also,of%20the%20Optimizely%20SDK%20library.) and [views](https://github.com/bbc/simorgh/tree/latest/src/app/hooks/useViewTracker#:~:text=A%20view%20event%20is%20also,of%20the%20Optimizely%20SDK%20library.), you can get the Optimizely object that you pass into these hooks from context https://github.com/optimizely/react-sdk?tab=readme-ov-file#example-2
```
<OptimizelyArticleCompleteTracking />
<OptimizelyPageViewTracking />
useOptimizelyScrollDepth();
```

**Pull request merged**
